ERRORS.md

```diff
@@ -0,0 +1,59 @@
+# Error Propagation in Broccoli & Ember CLI
+
+## Quick Summary
+
+This is a short summary of how errors during build time are surfaced to the browser. Let's look at a particular example.
+
+Let's say you have the following template:
+
+```handlebars
+<p>Sup you guise<p>
+```
+
+There's an unclosd `p` tag. Seems bad.
+
+Here's what is going to happen:
+
++ `ember-template-compiler` throws an exception
+  + `location`
+  + `message`
+  + `stack`
+
+Error contains information of where in the file error occured and what kind of error it was.
+
++ `ember-cli-htmlbars` will catch the error from `ember-template-compiler`.
+
+Because `ember-cli-htmlbars` is based on `broccoli-persistent-filter`, technically `broccoli-persistent-filter`
+will catch the exception first. For example, we attach `file` and `treeDir` information to errors [here](https://github.com/stefanpenner/broccoli-persistent-filter/blob/master/index.js#L283-L288).
+
++ `broccoli-builder` catches the error
+
+Ember CLI uses `broccoli-builder` to build its trees so if `broccoli-builder` throws an error,
+Ember CLI is aware and can act accordingly.
+
+At this level, we can attach more information to the error, like `broccoli` node/annotation.
+
++ `broccoli-middleware` catches the error
+
+This is where we return an error page for the browser. Given all the information, we get from `ember-cli-htmlbars`
+and `broccoli-builder`, we show an error page.
+
+## Error object
+
+As long as add-ons respect the following error contract, Ember CLI should be able to show useful error messages.
+Note, that is just an interface, not an actual class.
+
+```typescript
+interface Position {
+  line: number;
+  column: number;
+}
+
+interface BuildError {
+  stack: string;
```

Comment:
Maybe add some JSDOC style comments for each property with a bit more details? I'm not sure each peoperty needs fleshing out, but some do (e.g. "What the heck is a 'code frame'?").